xcpmd/src/acpi-events.c

```diff
@@ -30,133 +30,406 @@
 
     notify_com_citrix_xenclient_xcpmd_battery_info_changed(xcdbus_conn, XCPMD_SERVICE, XCPMD_PATH);
 }
 
-static void handle_pbtn_pressed_event(void)
-{
+
+static void handle_battery_status_event(int battery_index) {
+
+    char path[256];
+
+    //xcpmd_log(LOG_DEBUG, "Status change event on battery %d\n", battery_index);
+
+    update_battery_status(battery_index);
+    write_battery_status_to_xenstore(battery_index);
+
+    sprintf(path, "%s%i/%s", XS_BATTERY_EVENT_PATH, battery_index, XS_BATTERY_STATUS_EVENT_LEAF);
+    xenstore_write("1", path);
+
+    //Here for compatibility--should eventually be removed
+    xenstore_write("1", XS_BATTERY_STATUS_CHANGE_EVENT_PATH);
+
+    notify_com_citrix_xenclient_xcpmd_battery_status_changed(xcdbus_conn, XCPMD_SERVICE, XCPMD_PATH);
+}
+
+
+static void handle_lid_event(int status) {
+
+    int lid_status;
+
+    xcpmd_log(LOG_INFO, "Lid change event\n");
+
+    //We may have to check the sysfs if the ACPI string didn't tell us the status.
+    if (status == LID_UNKNOWN)
+        lid_status = get_lid_status();
+    else
+        lid_status = status;
+
+    xenstore_write(lid_status == LID_OPEN ? "open" : "closed", XS_LID_STATE_PATH);
+    xenstore_write("1", XS_LID_EVENT_PATH);
+    //notify_com_citrix_xenclient_xcpmd_lid_changed(xcdbus_conn, XCPMD_SERVICE, XCPMD_PATH);
+}
+
+
+static void handle_power_button_event(void) {
+
     xcpmd_log(LOG_INFO, "Power button pressed event\n");
-    xenstore_write("1", XS_PBTN_EVENT_PATH);
+    xenstore_write("1", XS_PWRBTN_EVENT_PATH);
     notify_com_citrix_xenclient_xcpmd_power_button_pressed(xcdbus_conn, XCPMD_SERVICE, XCPMD_PATH);
 }
 
-static void handle_sbtn_pressed_event(void)
-{
+
+static void handle_sleep_button_event(void) {
+
     xcpmd_log(LOG_INFO, "Sleep button pressed event\n");
-    xenstore_write("1", XS_SBTN_EVENT_PATH);
+    xenstore_write("1", XS_SLPBTN_EVENT_PATH);
     notify_com_citrix_xenclient_xcpmd_sleep_button_pressed(xcdbus_conn, XCPMD_SERVICE, XCPMD_PATH);
 }
 
-static void handle_bcl_event(enum BCL_CMD cmd)
-{
-    int ret, err;
 
-    if ( cmd == BCL_UP )
-    {
+static void handle_suspend_button_event(void) {
+
+    xcpmd_log(LOG_INFO, "Suspend button pressed event\n");
+    xenstore_write("1", XS_SUSPBTN_EVENT_PATH);
+    //notify_com_citrix_xenclient_xcpmd_suspend_button_pressed(xcdbus_conn, XCPMD_SERVICE, XCPMD_PATH);
+}
+
+
+static void handle_bcl_event(enum BCL_CMD cmd) {
+
+    if (cmd == BCL_UP) {
         xenstore_write("1", XS_BCL_CMD);
-	adjust_brightness(1, 0);
+        xenstore_write("1", XS_BCL_EVENT_PATH);
+        adjust_brightness(1, 0);
     }
-    else if ( cmd == BCL_DOWN )
-    {
+    else if (cmd == BCL_DOWN) {
         xenstore_write("2", XS_BCL_CMD);
-	adjust_brightness(0, 0);
+        xenstore_write("1", XS_BCL_EVENT_PATH);
+        adjust_brightness(0, 0);
+    }
+    else if (cmd == BCL_CYCLE) {
+        //xenstore_write("3", XS_BCL_CMD);
+        //xenstore_write("1", XS_BCL_EVENT_PATH);
```

Comment:
The new version of acpid has the ability to listen for a BCL_CYCLE event, but I wasn't sure exactly who would be consuming the information and if they would support the new event. This was left as a note that the capability is there if we want to go ahead and implement support for this on the other end.